.travis.yml

```diff
@@ -1,43 +1,46 @@
-language: python
+language: 
+  - python
+
 python:
-  - 2.7
-  - 3.6
-  - 3.7
-  - 3.8
+  - "3.6"
+  - "3.7"
+  - "3.8"
+  - "3.9"
+
 before_install:
   - sudo apt-get update
-  - if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
-      wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh -O miniconda.sh;
-    else
-      wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
-    fi
+  - wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
   - chmod +x miniconda.sh
   - ./miniconda.sh -b -p $HOME/miniconda
   - source "$HOME/miniconda/etc/profile.d/conda.sh"
   - conda init bash
   - conda config --set always_yes yes --set changeps1 no
   - conda config --add channels conda-forge
   - conda update -q conda
+
 install:
   # NOTE : conda-forge removed the .pc files for blas, cblas, lapack, and
   # lapacke inadvertently. Setting lapack < 3.9.0 ensures the .pc files are
   # available. See this issue:
   # https://github.com/conda-forge/blas-feedstock/issues/58
-  - conda create -q -n test-env python=$TRAVIS_PYTHON_VERSION "lapack<3.9.0" numpy cython ipopt future six setuptools sphinx numpydoc pkg-config
+  - conda create -q -n test-env python=$TRAVIS_PYTHON_VERSION
   - conda activate test-env
+  - conda install "lapack<3.9.0"
+  - conda install -y -q --file requirements.txt
+
 before_script:
-  - conda info
+  - conda info -a
   - conda list
   - pip --version
   - pip freeze
+
 script:
   # Make sure the library installs.
   - python setup.py install
-  - python -c "import ipopt"
-  - python test/examplehs071.py
-  - python test/exception_handling.py
-  - python test/lasso.py
-  - conda install scipy
-  - python test/rosen.py
-  # Make sure the docs build.
-  - cd doc && make html && cd ..
+  - python -c "import cyipopt"
+  # Run the Pytest tests.
+  - conda install -y -q pytest>=3.3.2
+  - pytest
```

Comment:
Scipy was installed as a second second to ensure the option aspect of it functioned. Do you have something that ensures cyipopt can operate without scipy?

Reply:
Addressed in https://github.com/matthias-k/cyipopt/pull/75/commits/38b855638492f7f870c55bff52bddba1ba56e46c, https://github.com/matthias-k/cyipopt/pull/75/commits/6d3d4532b10b8af101b321d07228abddc7562a3b, https://github.com/matthias-k/cyipopt/pull/75/commits/69f347ea79f9049fec616a617853ef7c68b4edbd and https://github.com/matthias-k/cyipopt/pull/75/commits/7fca6279bbdc1e2a4380b045e08884abbfda0d6d.

Reply:
I've added a new test module (`cyipopt/tests/unit/test_scipy_optional.py`) which contains two tests: one which asserts that `minimize_ipopt()` raises an `ImportError` if SciPy is not installed when called, and the other which asserts that `minimize_ipopt` functions correctly if SciPy in installed (by using the Rosenbrock example from the docs). `sys.modules` is used to check whether SciPy is/is not available and then conditionally skips the tests which aren't appropriate. The CI (both Travis and Appveyor) then run the tests once without SciPy installed, install SciPy and the rerun the tests (this way the non-SciPy-dependent tests are run with and without SciPy installed).

I'm not sure this is the cleanest solution (especially if the tests are being run locally) as which branch of the conditional tests are run is dependent on whether you have SciPy installed or not, but the other solutions on stack overflow (e.g. https://stackoverflow.com/questions/51044068/test-for-import-of-optional-dependencies-in-init-py-with-pytest-python-3-5/51048604#51048604) were causing problem when I tried their approach. I think this is an acceptable approach for now but could definitely be improved in the future.

Reply:
Ok, sounds good. Thanks for the explanation.